nixos/modules/services/web-apps/writefreely.nix

```
@@ -0,0 +1,102 @@
+{config, lib, pkgs, ...}:
+
+with lib;
+
+let
+  cfg = config.services.writefreely;
+  configFile = pkgs.writeText "config.ini" (generators.toINI {} cfg.settings);
+  settingsFormat = pkgs.formats.json {};
+in {
+  options.services.writefreely = {
+    enable = mkEnableOption "WriteFreely, a clean, Markdown-based publishing platform made for writers. Write together, and build a community.";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.writefreely;
+      description = ''
+        Overridable attribute of the WriteFreely package to use.
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "writefreely";
+      description = "User account under which writefreely runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "writefreely";
+      description = "User account under which writefreely runs.";
+    };
+
+    dataDir = mkOption {
+      default = "/var/lib/writefreely";
+      type = types.str;
```

```suggestion
      type = types.path;
```
otherwise it's possible to specify relative paths, which will almost certainly not work out as planned